File: clients/elixir/examples/onboarding_course.exs
alias PelagoDB.Client

endpoint = System.get_env("PELAGO_ENDPOINT", "127.0.0.1:27615")
database = System.get_env("PELAGO_DATABASE", "default")
namespace = System.get_env("PELAGO_NAMESPACE", "onboarding.demo")

client = Client.new(endpoint: endpoint, database: database, namespace: namespace)

schema = %{
  "name" => "Person",
  "properties" => %{
    "name" => %{"type" => "string", "required" => true},
    "role" => %{"type" => "string", "index" => "equality"},
    "level" => %{"type" => "int", "index" => "range"}
  }
}

{:ok, _} = Client.register_schema_dict(client, schema)

{:ok, node} =
  Client.create_node(client, "Person", %{
    "name" => "Ex Onboard",
    "role" => "Platform Engineer",
    "level" => 2
  })

{:ok, updated} = Client.update_node(client, "Person", node.id, %{"level" => 3})

IO.puts("elixir created node id=#{node.id}")
IO.puts("elixir updated node id=#{updated.id} level=3")

{:ok, stream} = Client.find_nodes(client, "Person", "level >= 3", 10)
IO.puts("elixir find level >= 3")

Enum.each(stream, fn item ->
  if item.node do
    IO.puts("- #{item.node.id} #{item.node.entity_type}")
  end
end)

{:ok, pql_stream} =
  Client.execute_pql(client, "Person @filter(level >= 3) { uid name role level }")

IO.puts("elixir pql level >= 3")

Enum.each(pql_stream, fn item ->
  if item.node do
    IO.puts("- #{item.node.id}")
  end
end)

File: clients/python/examples/onboarding_course.py
from __future__ import annotations

import os

from pelagodb import PelagoClient


def main() -> None:
    endpoint = os.getenv("PELAGO_ENDPOINT", "127.0.0.1:27615")
    database = os.getenv("PELAGO_DATABASE", "default")
    namespace = os.getenv("PELAGO_NAMESPACE", "onboarding.demo")

    client = PelagoClient(endpoint, database=database, namespace=namespace)

    person_schema = {
        "name": "Person",
        "properties": {
            "name": {"type": "string", "required": True},
            "role": {"type": "string", "index": "equality"},
            "level": {"type": "int", "index": "range"},
        },
    }
    client.register_schema_dict(person_schema)

    node = client.create_node(
        "Person",
        {
            "name": "Py Onboard",
            "role": "Analyst",
            "level": 3,
        },
    )
    updated = client.update_node("Person", node.id, {"level": 4})

    print(f"python created node id={node.id}")
    print(f"python updated node id={updated.id} level=4")

    print("python find level >= 3")
    for row in client.find_nodes("Person", "level >= 3", limit=10):
        print(f"- {row.id} {row.entity_type}")

    print("python pql level >= 3")
    for result in client.execute_pql("Person @filter(level >= 3) { uid name role level }"):
        if result.node:
            print(f"- {result.node.id}")

    client.close()


if __name__ == "__main__":
    main()
